frontend/src/components/doctor/DetailDoctorPage.tsx

```diff
@@ -10,133 +10,112 @@ import {
   Stack,
   Button,
   Flex,
+  Spinner,
 } from "@chakra-ui/react";
-import { fetchIndividualDoctorInfo } from "../api/FetchData";
 
-export type DoctorDetail = {
-  id: number;
-  first_name: string;
-  last_name: string;
-  email: string;
-  gender: string;
-  location: string;
-  insurance: string[];
-  days_available: string[];
-  speciality: string;
-  about: string;
-};
+import { Doctor } from "./DoctorControl"; // type import
+
+// export default function DetailDoctorPage({
+//   individualDoctor,
+// }: {
+//   individualDoctor?: Doctor;
+// })
 
 export default function DetailDoctorPage() {
+  let [individualDoctor, setIndividualDoctor] = useState<Doctor>();
+
   const params = useParams();
-  if (!params.id) {
-    return <div>No Doctor found</div>;
-  }
-  const doctorID = parseInt(params.id, 10);
 
-  const state = fetchIndividualDoctorInfo(doctorID);
-  if (!state) {
-    return <div>Doctor not found</div>;
-  }
+  // ! means I know doctor id always exist -- don't give error
+  const userId = parseInt(params.id!, 10);
```

Comment:
Does `params.id` always exist? It's better to check rather than suppress a potential error.